app/src/activity.c

```diff
@@ -19,18 +19,7 @@ LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
 #include <zmk/events/sensor_event.h>
 
 #include <zmk/activity.h>
-
-#if IS_ENABLED(CONFIG_USB_DEVICE_STACK)
-#include <zmk/usb.h>
-#endif
-
-bool is_usb_power_present(void) {
-#if IS_ENABLED(CONFIG_USB_DEVICE_STACK)
```

Comment:
Why is this being removed? It can be used in conjunction with charging detection to detect when it's fully charged. If USB power is present but board is not charging then it can be inferred that the board is fully charged

Reply:
you're right! didn't think of that. I only removed it because nobody was calling it (and it wasn't declared in a header anywhere LOL). now that you mention it, for the usecase i have in mind (charging but not via usb), i also want this distinction.

so i feel like we would want `is_charging` *and* `is_externally_powered` statuses?

Reply:
edit: will probably wait for the zephyr 3.5 update and #2154 to be merged before continuing on this

Reply:
okay, i ended up removing this for a few reasons:
1. `zmk_is_externally_powered` subsumes this functionality, and covers both the battery-charging case and the usb-plugged-in case
2. nobody was actually calling this, and it wasn't declared in a header anywhere

for detecting fully-charged, you would be able to do `zmk_is_externally_powered() && !zmk_battery_is_charging()`.